providers/implementations/encode_decode/decode_pem2der.c

```diff
@@ -145,9 +145,13 @@ static int pem2der_decode(void *vctx, OSSL_CORE_BIO *cin, int selection,
     int objtype = OSSL_OBJECT_UNKNOWN;
     const char *data_structure = NULL;
 
-    if (read_pem(ctx->provctx, cin, &pem_name, &pem_header,
-                 &der, &der_len) <= 0)
-        return 0;
+    ERR_set_mark();
+    ok = read_pem(ctx->provctx, cin, &pem_name, &pem_header,
+                  &der, &der_len) > 0;
+    ERR_pop_to_mark();
```

Comment:
Similarly to above with the read_der() I'd think that not being able to read any data from cin should be treated as fatal error.